security/permissions/src/main/java/io/quarkus/ts/openshift/security/basic/permissions/beanparam/permissions/NestedBeanParamPermission.java

```diff
@@ -0,0 +1,44 @@
+package io.quarkus.ts.openshift.security.basic.permissions.beanparam.permissions;
+
+import java.security.BasicPermission;
+import java.security.Permission;
+
+public class NestedBeanParamPermission extends BasicPermission {
+
+    private final String id;
+    private final String name;
+    private final String resourceId;
+    private final String resourceType;
+    private final String principalName;
+
+    public NestedBeanParamPermission(String permissionName, String id, String name,
+            String resourceId, String resourceType, String principalName) {
+        super(permissionName);
+        this.id = id;
+        this.name = name;
+        this.resourceId = resourceId;
+        this.resourceType = resourceType;
+        this.principalName = principalName;
+    }
+
+    @Override
+    public boolean implies(Permission permission) {
+
+        if (permission instanceof PossessedPermission possessedPermission) {
```

Comment:
This entire decision structure is hard to read. Please change it, so it does not have soo many nested blocks and several "return false" clauses.

Reply:
OK I've tweaked it